File: px-addons/src/main/java/com/mercadopago/android/px/addons/internal/TrackingDefaultBehaviour.java
package com.mercadopago.android.px.addons.internal;

import android.support.annotation.NonNull;
import com.mercadopago.android.px.addons.TrackingBehaviour;
import com.mercadopago.android.px.addons.model.Track;
import java.util.Map;

public class TrackingDefaultBehaviour implements TrackingBehaviour {

    @Override
    public void setApplicationContext(@NonNull final String applicationContext) {
        //Do nothing
    }

    @Override
    public void onView(@NonNull final String path, @NonNull final Map<String, ?> data) {
        //Do nothing
    }

    @Override
    public void onEvent(@NonNull final String path, @NonNull final Map<String, ?> data) {
        //Do nothing
    }

    @Override
    public void track(@NonNull final Track track) {
        //do nothing
    }
}

File: px-addons/src/main/java/com/mercadopago/android/px/addons/internal/TrackingDefaultBehaviour.kt
package com.mercadopago.android.px.addons.internal

import com.mercadopago.android.px.addons.TrackingBehaviour
import com.mercadopago.android.px.addons.model.Track
import com.mercadopago.android.px.addons.tracking.TrackerWrapper
import java.util.*

object TrackingDefaultBehaviour : TrackingBehaviour {

    private var trackerWrappers: List<TrackerWrapper>? = null

    fun setTrackWrapper(trackerWrapper: TrackerWrapper) {
        trackerWrappers = Collections.singletonList(trackerWrapper)
    }

    override fun track(track: Track) {
        trackerWrappers?.let { track.send(it) }
    }
}